### PR TITLE
Image style

### LINKED ISSUE #5: Add Style
#### As a user I want to see a visually pleasing application.

__Story__:

Our web application is really starting taking shape. We have added a few
features that provide value to our customers, but some of our customers are
complaining that the website is a quite bland. It is time to spice it up by
applying a visually appealing, and easy to maintain user interface to our web
application.

__Acceptance criteria__:
- [ ] The application has a header with navigation
- [ ] Images are displayed in an aesthetically pleasing list of cards
- [ ] Form errors are shown inline with the related input field
- [ ] I know how to use Bootstrap components like buttons and navbars in my
  application.
- [ ] I know the very basics of CSS and SCSS
- [ ] My SCSS files are properly named and organized using BEM convention
- [ ] My app has a "sticky" footer.
- [ ] My app has a few customized styles using bootstrap variable overrides.
- [ ] My app is responsive and is usable on a phone.
- [ ] The active link in the navigation header is differentiated.
- [ ] I use a bootstrap "container" to center my app contents (not necessarily "fluid").

__Reference material__:
- http://v4-alpha.getbootstrap.com/getting-started/introduction/
- https://github.com/plataformatec/simple_form
- http://getbem.com/naming/

__Dependencies__:
- Image Index

### PR DESCRIPTION
**Pull request by ltewksbury**
Closes #5

---

The following is a checklist to remind you of the steps to take when opening a
new PR. The horizontal line above and everything below can be removed from the
pull request description.

- [ ] I have replaced `X` above with the associated issue number.
- [ ] I have changed the issue's label to "pull request".
- [ ] I have assigned the designated reviewer to this PR.
- [ ] I have set the appropriate label on this PR.
- [ ] I have written clear commit messages ([ref](http://chris.beams.io/posts/git-commit/)).

If needed, please review the document on the
[Image Sharing Project](https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project)
for information on how labels should be used.

### CODE REVIEW

File: app/assets/stylesheets/images.scss [hunk 1]
```diff
@@ -1,3 +1,21 @@
...
+}
+
+.image-form {
+  width: 50%;
+}
+
+img {
+  width: 100%;
+  height: auto;
+}
```
Comment (ltewksbury) [inline app/assets/stylesheets/images.scss]: Goal:  When you click on one image, you see it without having to scroll.  This code works (mostly) for images where width > height.  It does not work for images where height > width.  I think I need something like if( width > height) { width: 100% and height: auto } else { height: 100% and width: auto }, but I don't know how to do that.  Thoughts?

File: app/views/images/_nav_header.html.erb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      <li class="nav-item">
+        <a class="nav-link active" href=<%= new_image_path %> >New Photo</a>
+      </li>
+      <li class="nav-item">
+        <a class="nav-link active" href=<%= images_path %> >Show All</a>
+      </li>
+      <% end %>
+    </ul>
+  </div>
+</div>
```
Comment (ltewksbury) [inline app/views/images/_nav_header.html.erb]: Please disregard this - fixed in a later submission.

File: app/views/images/index.html.erb [hunk 3]
```diff
@@ -1,21 +1,28 @@
...
-      <tr>
-        <td><%= link_to image_tag(image.image_url, width: 400), image %></td>
-      </tr>
+<body>
+<% rev_images = @images.reverse %>
+<% rev_images.each_slice(3).each do |image_group| %>

+<div class="album py-5">
+  <div class="container">
+    <div class="row">
```
Comment (sashaarkhipov) [inline app/views/images/index.html.erb]: How about replacing ` if image_group[0].present? then`,  `if image_group[1].present? then`, and ` if image_group[2].present? then` with the loop? Something like that:
```
<% image_group.each_with_index do |image, index| %>
  <% if image_group[index].present? %>
    <%= render "image_card", img: image_group[index] %>
  <% end %>
<% end %>
```

File: app/views/images/show.html.erb [hunk 4]
```diff
@@ -1,13 +1,15 @@
...
+      <%= image_tag(@image.image_url) %>
+    </div>
+  </body>
 
-<div class="field">
-  <p class="control">
-    <%= link_to 'Show All', images_path, class: 'button is-primary' %>
-  </p>
-</div>
+  <footer>
```
Comment (sashaarkhipov) [inline app/views/images/show.html.erb]: Is `<footer>` needed here?

**Pull request merged**